src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheck.java

```diff
@@ -113,6 +113,11 @@
      */
     private int lambdaStatementEnd = -1;
 
+    /**
+     * Hold the line-number where the last resource variable statement ended.
+     */
+    private int lastVariableResourceStatementEnd = -1;
```

Comment:
If you store information for 1 file, it needs to be erased for the next file.
You need to reset this variable in `beginTree`.

Reply:
done